styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -384,23 +392,27 @@ private void inspectPod(Watcher.Action action, Pod pod) {
     final RunState runState = stateManager.get(workflowInstance);
     if (runState == null) {
       LOG.warn("Pod event for unknown or inactive workflow instance {}", workflowInstance);
-      return;
+      return Optional.empty();
     }
 
     final Optional<String> executionIdOpt = runState.data().executionId();
     if (!executionIdOpt.isPresent()) {
       LOG.warn("Pod event for state with no current executionId: {}", podName);
-      return;
+      return Optional.empty();
     }
 
     final String executionId = executionIdOpt.get();
     if (!podName.equals(executionId)) {
       LOG.warn("Pod event not matching current exec id, current:{} != pod:{}",
           executionId, podName);
-      return;
+      return Optional.empty();
     }
 
-    final List<Event> events = translate(workflowInstance, runState, action, pod, stats);
+    return Optional.of(runState);
+  }
+
+  private void inspectPod(Watcher.Action action, Pod pod, RunState runState) {
+    final List<Event> events = translate(runState.workflowInstance(), runState, action, pod, stats);
```

Comment:
Didn't realise `translate` is so weird. It could get `workflowInstance` by itself.